src/styles/styled.js

```diff
@@ -20,28 +22,26 @@ const wrapper = StyledComponent =>
     );
   };
 
-const baseStyled = createStyled({wrapper, getInitialStyle, driver});
+type StyletronComponent<Props> = React.StatelessFunctionalComponent<Props> & {
+  // eslint-disable-next-line flowtype/no-weak-types
+  __STYLETRON__: any,
+};
+type WithTheme<Props> = {$theme: ThemeT} & Props;
+type StyleFn = {
+  (string, StyleObject): StyletronComponent<{}>,
```

Comment:
Could we also keep allowing

```js
type StyleFn = {
  (React.Component, StyleObject): StyletronComponent<{}>
```

so it matches the `styled` function from Styletron?

Reply:
I wrote a [comment](https://github.com/uber-web/baseui/pull/1241/files#diff-b919b7f3bd332ab6f0e816211d3f6561R35) below this line with thoughts on why I did not add this option. If you're interested, let's pair up so I can show you the strange inconsistencies I experienced

Reply:
based on in-person discussions it would be ideal to support this, but i'm not sure how to properly type it. does anyone have ideas?